metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_schema_terms.py

```diff
@@ -71,6 +71,15 @@ def extend_field(
         if len(terms_to_add) == 0:
             terms_to_add = all_terms
 
+        new_glossary_terms = []
+        new_glossary_terms.extend(server_terms)
+        new_glossary_terms.extend(terms_to_add)
+
+        unique_gloseary_terms = []
+        for term in new_glossary_terms:
+            if term not in unique_gloseary_terms:
+                unique_gloseary_terms.append(term)
+
```

Comment:
is this not basically not a set? Can we use set here?

Reply:
yes that is a set only, but the `GlossaryTermAssociationClass` is unhashable type; and hence we have to go by this way.